src/org/labkey/filetransfer/FileTransferManager.java

```diff
@@ -138,56 +159,100 @@ public TransferEndpoint getSourceEndpoint(ViewContext context)
             }
-            else
-                filter = new SimpleFilter(new SimpleFilter.InClause(FieldKey.fromParts(listDef.getKeyName()), selectedVals));
-            Sort sort = new Sort(FieldKey.fromParts(properties.get(REFERENCE_COLUMN)));
-            TableInfo tableInfo = listDef.getTable(context.getUser());
-            if (tableInfo != null)
+            return Collections.emptyList();
+        }
+
+
+        LOG.error("getFiles is returning nothing.  Either the user selected nothing and hit 'transfer', or something has gone terribly wrong.");
+        return Collections.emptyList();
+    }
+
+    //here is where we probably need to return a dataset or query definition instead of a list
+    @Nullable
+    public DatasetDefinition getMetadataList(Map<String, String> map)
+    {
+        if (map != null && !map.isEmpty())
+        {
+            String listContainerId = map.get(REFERENCE_FOLDER);
+            Container listContainer = ContainerManager.getForId(listContainerId);
+
+            if (listContainer != null)
             {
-                TableSelector selector = new TableSelector(tableInfo, Collections.singleton(properties.get(REFERENCE_COLUMN)), filter, sort);
-                return selector.getArrayList(String.class);
+                String listName = map.get(REFERENCE_DATASET);
+                Study study = StudyManager.getInstance().getStudy(listContainer);
+                if (study != null)
+                {
+                    DatasetDefinition dd = StudyManager.getInstance().getDatasetDefinitionByLabel(study, listName);
+                    return dd;
+                }
             }
         }
-        return Collections.emptyList();
+        LOG.error("getMetadataList is returning null ");
+        return null;
     }
 
     @Nullable
-    public ListDefinition getMetadataList(Map<String, String> map)
+    public Dataset getReferenceDataset(Map<String, String> map)
     {
         if (map != null && !map.isEmpty())
         {
             String listContainerId = map.get(REFERENCE_FOLDER);
             Container listContainer = ContainerManager.getForId(listContainerId);
+
             if (listContainer != null)
             {
-                String listName = map.get(REFERENCE_LIST);
-                return ListService.get().getList(listContainer, listName);
+                String datasetName = map.get(REFERENCE_DATASET);
+                Study study = StudyManager.getInstance().getStudy(listContainer);
+                if (study != null)
+                {
+                    Dataset refDataset = study.getDatasetByName(datasetName);
+
+                    if(refDataset != null)
+                    {
+                        return refDataset;
+                    }
+                    else
+                    {
+                        LOG.error("getReferenceDataset dataset was null (1)");
```

Comment:
It would be better to be more explicit in the error message here such as "Could not find dataset for name " + datasetName